### ISSUE TITLE
CucumberJS 2.0 API Migration

### ISSUE THREAD

**Issue by EliRobinson** [webdriverio/cucumber-boilerplate#107]
# Issue description
This boilerplate is really great, the only issue is it doesn't use the CucumberJS 2.0 API format. Would be great to have a version 2 branch which has API 2.0 references instead of API 1.0.

**e.g.**
[2.0](https://github.com/cucumber/cucumber-js/blob/master/docs/support_files/step_definitions.md) uses:
```
var {defineSupportCode} = require('cucumber');
var assert = require('assert');

defineSupportCode(function({Then, When}) {
  // Synchronous
  Then(/^Then the response status is (.*)$/, function (status) {
    assert.equal(this.responseStatus, status)
  });
```

while [1.0](https://github.com/cucumber/cucumber-js/blob/1.x/docs/support_files/step_definitions.md) uses:
```
module.exports = function () {
  // Synchronous
  this.Then(/^Then the response status is (.*)$/, function (status) {
    assert.equal(this.responseStatus, status)
  });
```

**Comment by wvankuipers** [comment]
We are depending on the [wdio-cucumber-framework](https://github.com/webdriverio/wdio-cucumber-framework) for support of the 2.0.0 framework, until that's implemented we can't support it.

**Comment by D34THWINGS** [comment]
I think you should re-open that since, they published a version with 2.3.1 of Cucumber in it

**Comment by EliRobinson** [comment]
+1

https://github.com/webdriverio/wdio-cucumber-framework/blob/master/package.json#L43

**Comment by wvankuipers** [comment]
See https://github.com/webdriverio/cucumber-boilerplate/pull/129

**Comment by EliRobinson** [comment]
Awesome! Thanks @wvankuipers - Will stay tuned

**Comment by wvankuipers** [comment]
Fixed in v1.3.0
